cstore_fdw.c

```diff
@@ -317,33 +351,33 @@ CStoreProcessUtility(Node * parseTree, const char *queryString,
 
 				if (strncmp(CSTORE_FDW_NAME, objectName, NAMEDATALEN) == 0)
 				{
-					removeCStoreDirectory = true;
+					removeCStoreRelationFiles = true;
 				}
 			}
 
-			CALL_PREVIOUS_UTILITY(parseTree, queryString, context, paramListInfo,
-								  destReceiver, completionTag);
-
-			if (removeCStoreDirectory)
+			if (removeCStoreRelationFiles)
 			{
-				RemoveCStoreDatabaseDirectory(MyDatabaseId);
+				cstoreTableList = CStoreTableList();
 			}
-		}
-		else
-		{
-			ListCell *fileListCell = NULL;
-			List *droppedTables = DroppedCStoreFilenameList((DropStmt *) parseTree);
 
-			CALL_PREVIOUS_UTILITY(parseTree, queryString, context, paramListInfo,
-								  destReceiver, completionTag);
+			CALL_PREVIOUS_UTILITY(parseTree, queryString, context,
+					paramListInfo, destReceiver, completionTag);
 
-			foreach(fileListCell, droppedTables)
+			if (cstoreTableList != NIL)
```

Comment:
If cstore_fdw extension is dropped, it goes down with pg_cstore_tables that contain relation ids for existing cstore tables. 

If drop extension command issued on the cstore_fdw extension first cstore relation ids are cached, process utility is called to execute drop extension command, if it is successfull, cached relation's files are marked for removal.

This had to be done here because drop trigger is not called when the extension is dropped.